HelloCharge-Swift/HelloChargeSwiftTableViewController.swift

```diff
@@ -57,40 +61,50 @@ class HelloChargeSwiftTableViewController: UITableViewController {
     // MARK: - UITableViewDataSource
 
     override func tableView(_ tableView: UITableView, willDisplay cell: UITableViewCell, forRowAt indexPath: IndexPath) {
-        if let section = Section(rawValue: indexPath.section) {
-            switch section {
-            case .supportedTenderTypes:
-                let tenderTypeForRow = allTenderTypes[indexPath.row]
-                let checked = supportedTenderTypes.contains(tenderTypeForRow)
-                cell.accessoryType = (checked ? .checkmark : .none)
-            case .settings:
-                if indexPath.row == 0 {
-                    cell.accessoryType = (clearsDefaultFees ? .checkmark : .none)
-                } else if indexPath.row == 1 {
-                    cell.accessoryType = (returnAutomaticallyAfterPayment ? .checkmark : .none)
-                }
-            case .amount, .optionalFields:
+        guard let section = Section(at: indexPath) else {
+            return
+        }
+
+        let showCheckmark: Bool
+        switch section {
```

Comment:
Might be nice to break this logic out into another method. Something like `shouldShowCheckmark(for indexPath: IndexPath)`